dotnetapp-selfcontained/README.md

```diff
@@ -1,42 +1,73 @@
 
-**Preparing your Environment**
+```console
+git clone https://github.com/dotnet/dotnet-docker-samples/
+cd dotnet-docker-samples/dotnetapp-selfcontained
+dotnet restore
+```
 
-1. Git clone this repository or otherwise copy this sample to your machine: `git clone https://github.com/dotnet/dotnet-docker-samples/`
-2. Navigate to this sample on your machine at the command prompt or terminal.
+Follow these steps to run the sample locally:
 
-**Build and Test the application locally**
+```console
+dotnet run Hello .NET Core from Docker
+```
 
-1. Restore dependencies: `dotnet restore`
-2. Run the application: `dotnet run`
-   - Alternatively, you can test run your application with the following two commands:
-   - `dotnet build`
-   - `dotnet bin/Debug/netcoreapp1.0/dotnetapp.dll`
+Follow these steps to build and run the sample locally as a self-contained app on Linux:
 
-**Publish the application**
+```console
+dotnet publish -c Release -o out
+./out/dotnetapp
+```
 
-1. Publish the application to ensure that all dependencies are included.
-   - If publishing for Debian, use the following command: `dotnet publish -c Release -r debian.8-x64 -o out`
-   - If publishing for Windows, use the following command: `dotnet publish -c Release -r win10-x64 -o out`
-   - Note, you can skip the `-r` argument if your current and target operating system are the same (or same enough).
-2 . Optionally, you can run/test the application from the publish directory using the following command: `dotnet out\dotnetapp.dll`. 
-   - This will only work if your current and target operating are the same (or same enough).
+Follow these steps to build and run the sample locally as a self-contained app on macOS Sierra:
 
-Note: Self-contained apps are operating system- and chip-specific after they are published. You can see that the Dockerfiles are configured to select specific assets produced by the publish verb. The -r argument instructs the publish command to select the correct native assets to include. See [.NET Core Runtime IDentifier (RID) catalog](https://docs.microsoft.com/dotnet/articles/core/rid-catalog) for more information.
+```console
+dotnet publish -c Release -o out -r osx.10.11-x64
+./out/dotnetapp
+```
 
-**Build and run Docker image**
+Note: The macOS Sierra instructions should be the same as the Linux instructions. Sierra is not yet fully supported, so requires workarounds until .NET Core is updated.
 
-1. Build the Docker image
-   - Commandline to create Docker image for Linux: `docker build -t dotnetbot .`
-   - Commandline to create Docker image for Nano: `docker build -t dotnetbot -f Dockerfile.nano .`
-2. Run the image: `docker run dotnetbot Hello .NET Core from Docker`
+Follow these steps to build and run the sample locally as a self-contained app on Windows:
+
+```console
+dotnet publish -c Release -o out
+out\dotnetapp.exe
+```
+
+Follow these steps to run this sample in a Linux container:
+
+```console
+dotnet restore
```

Comment:
`dotnet restore` is already specified above.  Maybe we should remove it from all OS steps or add it to all and remove the one in the clone steps at the top.